kubetest/azure.go

```diff
@@ -0,0 +1,348 @@
+	if err != nil {
+		return err
+	}
+	if string(o) != *acsEngineMD5 {
+		return fmt.Errorf("Wrong md5 sum for acs-engine")
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("unable to get current directory: %v", err)
+	}
+	log.Printf("Extracting tar file %v into directory %v", f.Name(), cwd)
+
+	if err = control.FinishRunning(exec.Command("tar", "-xzf", f.Name(), "--strip", "1")); err != nil {
+		return err
+	}
+	c.acsEngineBinaryPath = path.Join(cwd, "acs-engine")
+	return nil
+
+}
+
+func (c Cluster) generateARMTemplates() error {
+	err := os.Chdir(c.outputDir)
+	if err != nil {
+		return fmt.Errorf("Failed to change dir to %v: %v", c.outputDir, err)
+	}
+	if err := control.FinishRunning(exec.Command(c.acsEngineBinaryPath, "generate", "kubernetes.json", "--output-directory", c.outputDir)); err != nil {
+		return fmt.Errorf("Failed to generate ARM templates: %v", err)
+	}
+	return nil
+}
+
+func (c *Cluster) loadARMTemplates() error {
+	var err error
+	template, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM template file: %v", err)
+	}
+	err = json.Unmarshal([]byte(template), &c.templateJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall template %v", err.Error())
+	}
+
+	parameters, err := ioutil.ReadFile(path.Join(c.outputDir, "azuredeploy.parameters.json"))
+	if err != nil {
+		return fmt.Errorf("Error reading ARM parameters file: %v", err)
+	}
+	err = json.Unmarshal([]byte(parameters), &c.parametersJSON)
+	if err != nil {
+		return fmt.Errorf("Error unmarshall parameters %v", err.Error())
+	}
+	return nil
+}
+
+func (c *Cluster) getARMClient(ctx context.Context) error {
+	env, err := azure.EnvironmentFromName("AzurePublicCloud")
+	var client *AzureClient
+	if client, err = getAzureClient(env,
+		c.subscriptionId,
+		c.clientId,
+		c.tenantId,
+		c.clientSecret); err != nil {
+		return fmt.Errorf("Error trying to get Azure Client: %v", err)
+	}
+	c.azureClient = client
+	return nil
+}
+
+func (c *Cluster) createCluster() error {
+	var err error
+	err = c.getARMClient(c.ctx)
+	if err != nil {
+		return fmt.Errorf("Failed to generate ARM client: %v", err)
+	}
+	kubecfgDir, _ := ioutil.ReadDir(path.Join(c.outputDir, "kubeconfig"))
+	kubecfg := path.Join(c.outputDir, "kubeconfig", kubecfgDir[0].Name())
+	fmt.Printf("Setting kubeconfig env variable: kubeconfig path: %v", kubecfg)
+	os.Setenv("KUBECONFIG", kubecfg)
+	_, err = c.azureClient.EnsureResourceGroup(c.ctx, c.resourceGroup, c.location, nil)
+	if err != nil {
+		return fmt.Errorf("Could not ensure resource group: %v", err)
+	}
+	if _, err := c.azureClient.ValidateDeployment(
+		c.ctx, c.resourceGroup, c.name, c.templateJSON, c.parametersJSON,
+	); err != nil {
+		return fmt.Errorf("Template invalid: %v", err)
+	}
+	if _, err := c.azureClient.DeployTemplate(
+		c.ctx,
+		c.resourceGroup,
+		c.name,
+		c.templateJSON,
+		c.parametersJSON); err != nil {
```

Comment:
c.parametersJSON cannot be used directly. Should use its 'parameters' property instead.

Or it will throw:
```
The request content was invalid and could not be deserialized: 'Error converting value \"http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#\" to type 'Microsoft.WindowsAzure.ResourceStack.Frontdoor.Data.Definitions.DeploymentParameterDefinition'. Path 'properties.parameters.$schema'
```